common/functions/src/scalars/maths/abs.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use common_arrow::arrow::array::PrimitiveArray;
+use common_arrow::arrow::compute::arity::unary;
+use common_datavalues::prelude::*;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::scalars::function::Function;
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+
+#[derive(Clone)]
+pub struct AbsFunction {
+    _display_name: String,
+}
+
+impl AbsFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(AbsFunction {
+            _display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic())
+    }
+}
+
+macro_rules! impl_abs_function {
+    ($column:expr, $data_type:expr, $type:ident) => {{
+        let array = $column.column().to_array()?.get_array_ref();
```

Comment:
Column::Constant is break by to_array().

we can use `to_minimal_array` and apply abs then resize_constant.

See `apply_arithmetic`

Reply:
TIL, i switched to `to_minimal_array`, thank you